### PR TITLE
fix: move bitswap msg counters out of encode/decode

### PR DESCRIPTION
**Pull request by Arqu**

### CODE REVIEW

**Review by rklaehn — COMMENTED** [review]
So what was wrong before?

**Comment by Arqu** [comment]
> So what was wrong before?

The counters were in the encode/decode functions which skewed results as we encode a lot more than we actually send.

**Comment by rklaehn** [comment]
> The counters were in the encode/decode functions which skewed results as we encode a lot more than we actually send.

But that is also valuable information... Anyway, looks good.

File: iroh-bitswap/src/lib.rs [hunk 1]
```diff
@@ -287,6 +288,7 @@ impl<S: Store> Bitswap<S> {
 
     fn receive_message(&self, peer: PeerId, message: BitswapMessage) {
         inc!(BitswapMetrics::MessagesReceived);
+        record!(BitswapMetrics::MessageBytesIn, message.encoded_len() as u64);
```
Comment (dignifiedquire) [inline iroh-bitswap/src/lib.rs:291]: this encodes the message again :/

File: iroh-bitswap/src/network.rs [hunk 2]
```diff
@@ -123,6 +123,10 @@ impl Network {
             for i in 1..=retries {
                 debug!("send:{}: try {}/{}", peer, i, retries);
                 let (s, r) = oneshot::channel();
+                record!(
```
Comment (dignifiedquire) [inline iroh-bitswap/src/network.rs:126]: this might fail, so we don't actually know if these are the bytes we sent

Comment (Arqu) [reply iroh-bitswap/src/network.rs:126]: I'm aware, but would rather over then under count here ie we streamed half a response and channel broke.

File: iroh-bitswap/src/lib.rs (hunk 1 above)
Comment (Arqu) [reply iroh-bitswap/src/lib.rs:291]: Ik :( Want me to extend BitswapMessage with a pre calculated len?

File: iroh-bitswap/src/network.rs (hunk 2 above)
Comment (dignifiedquire) [reply iroh-bitswap/src/network.rs:126]: fair

File: iroh-bitswap/src/lib.rs (hunk 1 above)
Comment (dignifiedquire) [reply iroh-bitswap/src/lib.rs:291]: maybe we could cache it?

**Comment by Arqu** [comment]
> > The counters were in the encode/decode functions which skewed results as we encode a lot more than we actually send.
> 
> But that is also valuable information... Anyway, looks good.

https://github.com/n0-computer/iroh/issues/378

**Review by dignifiedquire — APPROVED** [review]
lets get it in and worry about improving that encoding issue later

**Pull request merged**